protocol/x/clob/abci.go

```diff
@@ -197,8 +197,24 @@ func PrepareCheckState(
 	}
 
 	// 6. Get all potentially liquidatable subaccount IDs and attempt to liquidate them.
-	subaccountIds := daemonLiquidationInfo.GetLiquidatableSubaccountIds()
-	if err := keeper.LiquidateSubaccountsAgainstOrderbook(ctx, subaccountIds); err != nil {
+	liquidatableSubaccountIds := daemonLiquidationInfo.GetLiquidatableSubaccountIds()
+	subaccountsToDeleverage, err := keeper.LiquidateSubaccountsAgainstOrderbook(ctx, liquidatableSubaccountIds)
```

Comment:
thinking about this more - `subaccountsToDeleverage` here should mostly contain subaccounts with unfilled liquidations and need to be deleveraged at bankruptcy price (-ve TNC), right?

we do have `DaemonLiquidationInfo.NegativeTncSubaccounts`, maybe we can just use that? this will de-couple step 6&7 completely. wdyt?

Reply:
yes it does contain -ve TNC subaccounts, paired with a perpetual id that we tried to liquidate with. I think it probably makes sense to use `DaemonLiquidationInfo.NegativeTncSubaccounts` as you suggested. Unsure if we want to deleverage using the same perpetual we tried to liquidate with though, if so we would need to add some extra logic for that

Reply:
yeah maybe we can do this in a separate PR. no need to block on this

Reply:
maybe create a ticket and add a TODO?